services/github/github-gist-last-commit.service.js

```diff
@@ -0,0 +1,47 @@
+import Joi from 'joi'
+import { formatDate } from '../text-formatters.js'
+import { age as ageColor } from '../color-formatters.js'
+import { GithubAuthV3Service } from './github-auth-service.js'
+import { documentation, errorMessagesFor } from './github-helpers.js'
+
+const schema = Joi.object({
+  updated_at: Joi.string().required(),
+}).required()
+
+export default class GithubGistLastCommit extends GithubAuthV3Service {
+  static category = 'activity'
+  static route = { base: 'github/gists', pattern: ':gistId' }
+  static examples = [
+    {
+      title: 'GitHub Gist last commit',
+      namedParams: {
+        gistId: '8710649',
+      },
+      staticPreview: this.render({ commitDate: '2022-07-29T20:01:41Z' }),
+      keywords: ['latest'],
+      documentation,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'last commit' }
+
+  static render({ commitDate }) {
+    return {
+      message: formatDate(commitDate),
+      color: ageColor(Date.parse(commitDate)),
+    }
+  }
+
+  async fetch({ gistId }) {
+    return this._requestJson({
+      url: `/gists/${gistId}`,
+      schema,
+      errorMessages: errorMessagesFor('gist not found'),
+    })
+  }
+
+  async handle({ gistId }) {
+    const body = await this.fetch({ gistId })
+    return this.constructor.render({ commitDate: body.updated_at })
```

Comment:
This is perfectly fine but just note for future reference that destructuring may be an easier way of extracting+renaming, e.g.

```suggestion
    const { updated_at: commitDate } = await this.fetch({ gistId })
    return this.constructor.render({ commitDate })
```

Thought it's mostly stylistic

Reply:
I didn't know this way of destructuring and concurrent renaming, very neat - thanks, updated ;)